### ISSUE TITLE
Performance issues?

### ISSUE THREAD

**Issue by xnaas** [Kwoth/NadekoBot#987]
Hopefully I'm not the only one suffering performance issues after upgrading to 1.1.x. I had hoped upgrading to 1.1.2 would help, but it hasn't.

Performance is incredibly hit or miss with the bot for the past couple versions. If you type a message and then a command, it will fail to see the command. You'll have to run the command *again* to get the bot to respond to it.

Here's a recent example from a few minutes ago:
![img](http://i.imgur.com/mahnCHG.png)

You can see the user quickly sent `$race` and then `$jr`, but the bot did not see it the first time and the command had to be repeated. This is easily replicate-able behavior and while not bot-breaking, it is incredibly inconvenient. 😛 

Basically, wondering if anyone else is having any issues like this, if it's a known problem, etc. And before you ask if there's an issue with my server or proximity of the Discord server to the actual server, those are all non-issues. [Server](http://pcpartpicker.com/user/xnaas/saved/kQ86Mp) is severely over-spec'd. On a gigabit up and down connection. Home server is in Austin, Discord server location is US South.

**Comment by Kwoth** [comment]
read the patch notes in the future.

1.5s cooldown per command per user.

**Comment by xnaas** [comment]
Hmm...must've missed that one! I do read and love showing off when new stuff is added. 😄

**Comment by Kwoth** [comment]
some commands, like $slot, have internal cooldowns to themselves, usually specified in the description (-h commandname)

**Comment by xnaas** [comment]
I take it there's no way to remove the global cooldown? Skimmed through database briefly and saw the `CommandCooldown` table was empty. 😛

**Comment by xnaas** [comment]
Also...is there any reason that would affect if you send a message that is *not* a command and then a message that *is* a command, it still is affected by that 1.5s cooldown? Because I can't think of anything else that would be causing that particular problem consistently. 😄

**Comment by Kwoth** [comment]
i was just about to look into that, yesterday when testing something, it seemed it was taking any message and triggering the cooldown

**Comment by xnaas** [comment]
Glad to hear I'm not losing my mind (yet)! 😃

**Comment by xnaas** [comment]
Taking that as a no for disabling global cooldown then? 😛

**Comment by Kwoth** [comment]
i'll reduce it to 750ms on self hosted

**Comment by xnaas** [comment]
Good enough! Thanks for the responses, information, etc. 👍
